src/pages/dataset/[accession_id].astro

```diff
@@ -187,13 +196,100 @@ const total_file_type_list = get_file_types(study_info)
     </div>
   </section>
 
+  <section class="vf-intro | embl-grid embl-grid--has-centered-content | section-spacing">
+
+    <div class="vf-section-header">
+      <h2 class="vf-section-header__heading" id="file_references">
+        All files
+      </h2>
+    </div>
+    <div>All files of dataset
+      <select id="dataset_selector" class="form-select" data-dataset-uuids={JSON.stringify(dataset_uuids)}>
+        <option value="" disabled selected>Select a Dataset</option>
+      </select>
+      
+      <table id="file_references_table" class="display">
+      </table>
+    </div>
+  </section>
+
     <script>
       document.addEventListener("DOMContentLoaded", function () {
         $("#viewable_images_table").DataTable({
           scrollX: true,
           columnDefs: [{ type: "natural", targets: 0 }],
         });
       });
+      document.addEventListener("DOMContentLoaded", function () {
+        const dropdown = document.getElementById("dataset_selector");
+        const datasetDetails = JSON.parse(dropdown.dataset.datasetUuids); 
+
+        // Populate the dropdown menu with dataset details
+        datasetDetails.forEach(([uuid, fileCount, title]) => {
+            const option = document.createElement("option");
+            option.value = uuid;
+            option.text = `${title} (${fileCount} files)`; 
+            dropdown.appendChild(option);
+        });
+
+        let dataTable; 
+
+        // Function to initialize or update the DataTable
+        const initializeDataTable = (datasetUuid, totalFiles) => {
+            // set page size parameter of the api url with an upper limit of 1000
+            const pageSize = totalFiles < 1000 ? totalFiles : 1000;
```

Comment:
I'm not sure this is needed, if pageSize=1000 and there are only, say, 100 things to return, the api just returns those

Reply:
I wasn't sure how api works hence this line.